subsys/bluetooth/mesh/dk_prov.c

```diff
@@ -160,6 +160,31 @@ static const struct bt_mesh_prov prov = {
 
 const struct bt_mesh_prov *bt_mesh_dk_prov_init(void)
 {
+	/* Generate an RFC-4122 version 4 compliant UUID.
+	 * Format:
+	 *
+	 * 0                   1                   2                   3
+	 * 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 * |                          time_low                             |
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 * |       time_mid                |         time_hi_and_version   |
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 * |clk_seq_hi_res |  clk_seq_low  |         node (0-1)            |
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 * |                         node (2-5)                            |
+	 * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 *
+	 * Where the 4 most significant bits of time_hi_and_version shall be
+	 * 0b0010 and the 2 most significant bits of clk_seq_hi_res shall be
+	 * 0b10. The remaining fields have no required values, and are fetched
+	 * from the HW info device ID. The fields are encoded in big endian
+	 * format.
+	 *
+	 * https://tools.ietf.org/html/rfc4122
+	 */
 	hwinfo_get_device_id(dev_uuid, sizeof(dev_uuid));
+	dev_uuid[6] = (dev_uuid[6] & BIT_MASK(4)) | BIT(5);
+	dev_uuid[8] = (dev_uuid[8] & BIT_MASK(6)) | BIT(7);
```

Comment:
BIT_MASK(6) will cause 3 (6-1=5 bits will be 1) most significant bits to be set to zero (`0b0001_1111`). According to RFC4122 section 4.1.1, for RFC4122 variant bit6 is don't care. Although this is a minor issue, I think, BIT_MASK(7) is more appropriate here.

Reply:
I believe `BIT_MASK(6)` expands to `((1 << 6) - 1) = (64 - 1) = 0b111111`?

Reply:
Ok, right. I think I got confused due to the differences in the BIT() macro and BIT_MASK() macro.